server/files/api_views.py

```diff
@@ -21,14 +22,24 @@ def perform_destroy(self, instance):
 
```

Comment:
Since you're checking on the HTTP level for permissions, it makes sense to override this in the `destroy` method instead of the `perform_destroy` method:

```pytho
    def destroy(self, request, *args, **kwargs):
        instance = self.get_object()
        if instance.notebook.owner != request.user:
             raise PermissionDenied
        return super().destroy(request, *args, **kwargs)
```